examples/01-circular_buffer/.gitignore

```diff
@@ -0,0 +1,2 @@
+circular_buffer
```

Comment:
Is this here intentionally? :confused:

Reply:
`circular_buffer` is the compiled executable (required by the test). I thought it was convenient to add a `.gitignore` to avoid these files ever being accidentally committed, but of course it's not necessary. Up to you :slightly_smiling_face: